solardatatools/pvpro_post_processing.py

```diff
@@ -0,0 +1,509 @@
+""" PVPRO Post Processing Module
+
+This module contains a class that takes in the output dataframe of PVPRO and contains
+methods to process the dataset, perform signal decompositions to model degradation trends,
+analyze how well the models fit, and visualize the trends. 
+
+"""
+
+import numpy as np
+
+import pandas as pd
+import cvxpy as cp
+
+import matplotlib.pyplot as plt
+
+from collections import defaultdict
+from scipy.stats import mode
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler
+from time import time
+from solardatatools.utilities import progress
+
+
+class PVPROPostProcessor():
+    
+    def __init__(self, file_name, period, index_col=0, dates=None, verbose=False, bp=True, 
+                 bounds='Default', est=False):
+        # imports data and assigns index column and date columns - the optional dates entry
+        # should be a list of column indices [i, j, k]
+        # the verbose keyword argument indicates whether the % of points on the boundaries 
+        # should be printed
+        # the bounds keyword arguments allows the user to input custom boundary values; input
+        # is a dictionary
+        # est should only be True when the input dataframe contains 'ref' and 'ref_est'
+        # columns
+        
+        if dates is None:
+            dates = [0]
+        else:
+            dates = dates
+        
+        # attributes
```

Comment:
Attributes that are set by __init__ don't need to be "declared" as None earlier in the function. A typical structure for init would be 

- Run setup code snippets
- Set attributes that will be set by __init__
- "Declare" other attributes as None that will be set by other methods